jaeger-core/src/main/java/com/uber/jaeger/Configuration.java

```diff
@@ -327,6 +353,38 @@ public String getManagerHostPort() {
     }
   }
 
+  /**
+   * CodecConfiguration can be used to support additional trace context propagation codec.
+   */
+  public static class CodecConfiguration {
+    private List<Codec<TextMap>> codecs;
+
+    public CodecConfiguration(List<Codec<TextMap>> codecs) {
+      this.codecs = codecs;
+    }
+
+    public static CodecConfiguration fromEnv() {
+      List<Codec<TextMap>> codecs = new LinkedList<Codec<TextMap>>();
+      if (getPropertyAsBool(JAEGER_B3_CODEC)) {
+        codecs.add(new B3TextMapCodec());
+      }
+      return new CodecConfiguration(codecs);
+    }
+
+    public void config(Tracer.Builder builder) {
+      if (!codecs.isEmpty()) {
+        // Replace existing TEXT_MAP and HTTP_HEADERS codec with one that also includes the
+        // list of configured codecs
+        TextMapCodec textMapCodec = TextMapCodec.builder().withUrlEncoding(false).withCodecs(codecs).build();
+        builder.registerInjector(Format.Builtin.TEXT_MAP, textMapCodec);
+        builder.registerExtractor(Format.Builtin.TEXT_MAP, textMapCodec);
+        TextMapCodec httpCodec = TextMapCodec.builder().withUrlEncoding(true).withCodecs(codecs).build();
```

Comment:
one downside of this is the assumption that all 3rd party codecs work the same for TEXT_MAP vs. HTTP_HEADERS formats, which is not the case with our internal codec.